syntax/parse.go

```diff
@@ -938,3 +953,99 @@ func terminatesExprList(tok Token) bool {
 	}
 	return false
 }
+
+// Comment assignment.
+// We build two lists of all subexpressions, preorder and postorder.
+// The preorder list is ordered by start location, with outer expressions first.
+// The postorder list is ordered by end location, with outer expressions last.
+// We use the preorder list to assign each whole-line comment to the syntax
+// immediately following it, and we use the postorder list to assign each
+// end-of-line comment to the syntax immediately preceding it.
+
+// flattenAst returns the list of AST nodes, both in prefix order and in postfix
+// order.
+func flattenAst(f *File) (pre []Node, post []Node) {
```

Comment:
flattenAST

(Go uses the same case for all letters of an initialism.)

Reply:
Done